common/src/main/java/org/apache/sedona/common/raster/RasterBandAccessors.java

```diff
@@ -134,6 +135,25 @@ public static double[] getSummaryStats(GridCoverage2D raster) {
 //        return getSummaryStats(raster, 1, excludeNoDataValue);
 //    }
 
+    public static GridCoverage2D getBand(GridCoverage2D rasterGeom, int[] bands) throws FactoryException {
+        double[] metadata = RasterAccessors.metadata(rasterGeom);
+        GridCoverage2D resultRaster = RasterConstructors.makeEmptyRaster(bands.length, (int) metadata[2], (int) metadata[3],
+                metadata[0], metadata[1], metadata[4], metadata[5], metadata[6], metadata[7], (int) metadata[8]);
+        double[] curBandData;
+        Double noDataValue;
+        for (int curBand: bands) {
+            RasterUtils.ensureBand(rasterGeom, curBand);
+            curBandData = MapAlgebra.bandAsArray(rasterGeom, curBand);
+            noDataValue = RasterBandAccessors.getBandNoDataValue(rasterGeom, curBand);
+            if (noDataValue != null) {
+                resultRaster = MapAlgebra.addBandFromArray(resultRaster, curBandData, curBand, noDataValue);
```

Comment:
You should not directly use `addBandFromArray`. Each function call will copy and create a new raster, which is very slow. You should implement a version of `copyRasterAndReplaceBand` or borrow some logic from that to set pixel values of all bands in one go.

Reply:
Sure, I was going to do that but I thought to reuse.